src/engine/blocks.js

```diff
@@ -75,37 +75,34 @@ Blocks.prototype.getBranch = function (id, branchNum) {
     }
 
     // Empty C-block?
-    if (!(inputName in block.inputs)) return null;
-    return block.inputs[inputName].block;
+    var input = block.inputs[inputName];
+    return (typeof input === 'undefined') ? null : input.block;
 };
 
 /**
  * Get the opcode for a particular block
- * @param {?string} id ID of block to query
+ * @param {?object} block The block to query
  * @return {?string} the opcode corresponding to that block
  */
-Blocks.prototype.getOpcode = function (id) {
-    var block = this._blocks[id];
+Blocks.prototype.getOpcode = function (block) {
```

Comment:
Since you requested to remove the direct access to the opcode from this PR, I have refactored the 'getOpcode' function to work in the same way as the other functions in the block class such that you now pass in the block to be retrieved instead of the block ID. I note though that this still introduces a redundant `=== 'undefined'` check that isn't needed when we know the object is not undefined at the point of call?